src/client/imagefilters.cpp

```diff
@@ -65,28 +65,37 @@ class Bitmap {
 	}
 };
 
-/* Fill in RGB values for transparent pixels, to correct for odd colors
- * appearing at borders when blending.  This is because many PNG optimizers
- * like to discard RGB values of transparent pixels, but when blending then
- * with non-transparent neighbors, their RGB values will show up nonetheless.
- *
- * This function modifies the original image in-place.
- *
- * Parameter "threshold" is the alpha level below which pixels are considered
- * transparent. Should be 127 when the texture is used with ALPHA_CHANNEL_REF,
- * 0 when alpha blending is used.
- */
-void imageCleanTransparent(video::IImage *src, u32 threshold)
+template <bool IS_A8R8G8B8>
+static void imageCleanTransparentWithInlining(video::IImage *src, u32 threshold)
 {
-	core::dimension2d<u32> dim = src->getDimension();
+	void *const src_data = src->getData();
+	const core::dimension2d<u32> dim = src->getDimension();
+	const u32 src_pitch = src->getPitch();
+
+	auto get_pixel = [src, src_data, dim](u32 x, u32 y) -> video::SColor {
+		if constexpr (IS_A8R8G8B8) {
+			return reinterpret_cast<u32 *>(src_data)[y*dim.Width + x];
+		} else {
+			return src->getPixel(x, y);
+		}
+	};
+	auto set_pixel = [src, src_data, src_pitch, dim](u32 x, u32 y, video::SColor color) {
+		if constexpr (IS_A8R8G8B8) {
+			u32 *dest = reinterpret_cast<u32 *>(
+					reinterpret_cast<u8 *>(src_data) + (y * src_pitch) + (x << 2));
```

Comment:
checking IImage.h irrlicht guarantees that pitch = BPP * width
so you can use the same formula as above here

Reply:
Done.